src/pages/signin/ValidateCodeForm/BaseValidateCodeForm.js

```diff
@@ -196,6 +205,8 @@ function BaseValidateCodeForm(props) {
      * Clears local and Onyx sign in states
      */
     const clearSignInData = () => {
+        // Reset the user's preference for signing in with SAML
+        props.setIsUsingSAMLLogin(props.account.isSAMLEnabled || props.account.isSAMLRequired)
```

Comment:
Should we set false here, so when user enters new email, we set the state again?

Reply:
Oh, yep - i think you're right and false would work here

Reply:
Actually...  I think we want to retain this because if the user chooses the magic code one time we don't want to lock the into that option - the next time they try to log in they should be able to choose again whether they want SAML or the magic code